include/sys/util.h

```diff
@@ -426,6 +426,30 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 /** @brief Number of Hz in @p x MHz */
 #define MHZ(x) (KHZ(x) * 1000)
 
+/**
+ * @brief Wait for an expression to return true with a timeout
+ *
+ * Spin on an expression with a timeout and optional delay between iterations
+ *
+ * Commonly needed when waiting on hardware to complete an asynchronous
+ * request to read/write/initialize/reset, but useful for any expression.
+ *
+ * @param expr Trueth expression you wish to spin on, ex: XYZREG & XYZREG_EN
```

Comment:
Four items here.

The first is a typo. Trueth -> Truth
The second and third are more more style oriented. The first of which is that as a rule, I think we should avoid using 'you' in the documentation. The second has to do with ending a sentence with a preposition (on). 
Possible replacement: Truth expression upon which to poll

The fourth item, is admittedly really pedantic--I think we are looking for "e.g." as opposed to "ex"